### PR TITLE
python3Packages.{xdis,uncompyle6}: bump to latest to support python 3.12

### PR DESCRIPTION
**Pull request by melvyn2**
## Description of changes
- xdis: 6.1.0 provides support for python bytecode and runtime up to 3.12
- uncompyle6: 3.9.1 provides support for python runtime up to 3.12, and bytecode up to 3.8
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by onny** [comment]
Result of `nixpkgs-review pr 299654` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python311Packages.uncompyle6</li>
    <li>python311Packages.uncompyle6.dist</li>
    <li>python311Packages.xdis</li>
    <li>python311Packages.xdis.dist</li>
    <li>python312Packages.xdis</li>
    <li>python312Packages.xdis.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/uncompyle6/default.nix [hunk 1]
```diff
@@ -2,6 +2,7 @@
 , buildPythonPackage
 , fetchPypi
 , pythonAtLeast
```
Comment (onny) [inline pkgs/development/python-modules/uncompyle6/default.nix:4]: This dependency isn't required anymore

File: pkgs/development/python-modules/xdis/default.nix [hunk 2]
```diff
@@ -43,15 +36,19 @@ buildPythonPackage rec {
...
+    #   /build/source/pytest/test_disasm.py
+    # which is not the same as the test file we want to collect:
+    #   /build/source/test_unit/test_disasm.py
     "test_unit/test_disasm.py"
+
+    # Doesn't run on non-2.7 but has global-level mis-import
+    "test_unit/test_dis27.py"
   ];
 
+
```
Comment (onny) [inline pkgs/development/python-modules/xdis/default.nix:51]: duplicate new line

**Comment by melvyn2** [comment]
@onny are any further changes needed?

**Comment by onny** [comment]
Thank you :)

**Comment by wegank** [comment]
Not sure what happened, but this doesn't seem to fix `python3Packages.xdis`? https://hydra.nixos.org/job/nixpkgs/trunk/python311Packages.xdis.x86_64-linux

**Comment by melvyn2** [comment]
Python 3.11 was updated to 3.11.9 after the PR was merged (or maybe after it was created), #316214 should fix it. Not sure where it should be targeted though.

**Pull request merged**